python/pyspark/sql/tests.py

```diff
@@ -459,6 +459,23 @@ def filename(path):
         row2 = df2.select(sameText(df2['file'])).first()
         self.assertTrue(row2[0].find("people.json") != -1)
 
+    def test_udf_defers_judf_initalization(self):
```

Comment:
@holdenk Separate case it is. As long as implementation is correct an overhead is negligible.